NIPTool/API/external/api/api_v1/endpoints/batches.py

```diff
@@ -1,137 +1,187 @@
+def batches(
+    request: Request, adapter: NiptAdapter = Depends(get_nipt_adapter)
+):  # , user: User = Depends(get_current_active_user)):
     """List of all batches"""
 
-    all_batches = list(adapter.batches())
-    return templates.TemplateResponse("batches.html",
-                                      context={"request": request, "batches": all_batches,
-                                               "current_user": 'mayapapaya',
-                                               "page_id": "all_batches"})
+    all_batches = list(find.batches(adapter=adapter))
+    return templates.TemplateResponse(
+        "batches.html",
+        context={
+            "request": request,
+            "batches": all_batches,
+            "current_user": CURRENT_USER,
+            "page_id": "all_batches",
+        },
+    )
 
 
 @router.get("/{batch_id}/")
 def batch(request: Request, batch_id: str, adapter: NiptAdapter = Depends(get_nipt_adapter)):
     """Batch view with table of all samples in the batch."""
-    samples = adapter.batch_samples(batch_id)
-    return templates.TemplateResponse('batch/tabs/table.html',
-                                      context={"request": request, "batch": adapter.batch(batch_id),
-                                               "sample_info": [get_sample_info(sample) for
-                                                               sample in samples],
-                                               "page_id": "batches",
-                                               "current_user": 'mayapapaya'})
+    samples = find.batch_samples(batch_id=batch_id, adapter=adapter)
+    return templates.TemplateResponse(
+        "batch/tabs/table.html",
+        context={
+            "request": request,
+            "batch": find.batch(batch_id=batch_id, adapter=adapter),
+            "sample_info": [get_sample_info(sample) for sample in samples],
+            "page_id": "batches",
+            "current_user": CURRENT_USER,
+        },
+    )
+
+
+@router.post("/{batch_id}/")
+def batch(request: Request, batch_id: str, adapter: NiptAdapter = Depends(get_nipt_adapter)):
+    """Batch view with table of all samples in the batch."""
+    samples = find.batch_samples(batch_id=batch_id, adapter=adapter)
+    return templates.TemplateResponse(
+        "batch/tabs/table.html",
+        context={
+            "request": request,
+            "batch": find.batch(batch_id=batch_id, adapter=adapter),
+            "sample_info": [get_sample_info(sample) for sample in samples],
+            "page_id": "batches",
+            "current_user": CURRENT_USER,
+        },
+    )
 
 
 @router.get("/{batch_id}/{ncv}")
 def NCV(request: Request, batch_id: str, ncv, adapter: NiptAdapter = Depends(get_nipt_adapter)):
     """Batch view with with NCV plot"""
-
+    batch: Batch = find.batch(batch_id=batch_id, adapter=adapter)
+    u = get_tris_control_normal(adapter, ncv)
```

Comment:
Can we name this to something else than `u`?

Reply:
haha. yep we can :) 👍